nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BulkActionObject.java

```diff
@@ -0,0 +1,228 @@
+                    return repository;
+                }
+                return param;
+            }).toArray();
+        }
+
+        DocumentModel searchDocumentModel = getSearchDocumentModel(getContext().getCoreSession(), ppService,
+                pageProviderName, namedParameters);
+
+        Map<String, Serializable> props = Collections.singletonMap(CoreQueryDocumentPageProvider.CORE_SESSION_PROPERTY,
+                (Serializable) getContext().getCoreSession());
+
+        PageProvider<DocumentModel> pageProvider = (PageProvider<DocumentModel>) ppService.getPageProvider(
+                pageProviderName, searchDocumentModel, null, null, null, props, quickFilterList, parameters);
+
+        String quickFiltersClause = "";
+        for (QuickFilter quickFilter : quickFilterList) {
+            String clause = quickFilter.getClause();
+            if (!quickFiltersClause.isEmpty() && clause != null) {
+                quickFiltersClause = NXQLQueryBuilder.appendClause(quickFiltersClause, clause);
+            } else {
+                quickFiltersClause = clause != null ? clause : "";
+            }
+        }
+
+        String query;
+        def = pageProvider.getDefinition();
+        WhereClauseDefinition whereClause = def.getWhereClause();
+        if (whereClause == null) {
+            String originalPattern = def.getPattern();
+            String pattern = quickFiltersClause.isEmpty() ? originalPattern
+                    : StringUtils.containsIgnoreCase(originalPattern, " WHERE ")
+                            ? NXQLQueryBuilder.appendClause(originalPattern, quickFiltersClause)
+                            : originalPattern + " WHERE " + quickFiltersClause;
+
+            query = NXQLQueryBuilder.getQuery(pattern, pageProvider.getParameters(), def.getQuotePatternParameters(),
+                    def.getEscapePatternParameters(), searchDocumentModel, null);
+        } else {
+            if (searchDocumentModel == null) {
+                throw new NuxeoException(String.format(
+                        "Cannot build query of provider '%s': " + "no search document model is set", getName()));
+            }
+            query = NXQLQueryBuilder.getQuery(searchDocumentModel, whereClause, quickFiltersClause,
+                    pageProvider.getParameters(), null);
+        }
+
+        Map<String, Serializable> params = BulkParameters.paramsToMap(actionParams);
+
+        BulkCommand command = new BulkCommand().withAction(actionId)
+                                               .withRepository(repository)
+                                               .withUsername(username)
+                                               .withQuery(query)
+                                               .withParams(params);
+        String commandId = Framework.getService(BulkService.class).submit(command);
+        return Response.status(Response.Status.ACCEPTED).entity(commandId).build();
```

Comment:
Assuming the method is producing `application/json` you should return more than a `commandId`.
Anyway, we should return a JSON object here to avoid breaking the compatibility if later on we want to return more than just the `commandId`.